### DISCUSSION TITLE
this.$api is not recognized in typescript.

### DISCUSSION THREAD

**Discussion by lesjon** [quasarframework/quasar#15924] [Ideas / Proposals]
When i'm using typescript and use the api. i execute requests with this.$api.get... but typescript does not know the type of this.$api.

**Comment by metalsadman** [comment]
what is this.$api? and how are you using it? maybe you should just export and import it or use provide inject, see vue docs about it.

**Reply by lesjon** [reply]
Thanks for the quick reply, I created a [pull request](https://github.com/quasarframework/quasar/pull/15925), i dont know how to link it to this discussion.
this.$api is created in `src/boot/axios.ts` (see example in PR). There i also show how it could already be used.
